python/pysrc/kaskada/_timestream.py

```diff
@@ -880,6 +893,84 @@ def record(self, fields: Callable[[Timestream], Mapping[str, Arg]]) -> Timestrea
         """
         return record(fields(self))
 
+    def round(self) -> Timestream:
+        """Return a Timestream with all values rounded to the nearest integer.
+
+        Returns:
+            A Timestream of the same type as `self`. The result contains `null`
+              if the value was `null` at that position. Otherwise, it contains
+              the result of rounding the value to the nearest integer.
+
+        Notes:
+            This method may be applied to any numeric type. For anything other
+              than `float32` and `float64` it has no affect since the values
+              are already integers.
+
+        See Also:
+            `ceil` and `floor`
+        """
+        return Timestream._call("round", self)
+
+    def sqrt(self) -> Timestream:
+        """Return a Timestream with the square root of all values.
+
+        Returns:
+            A Timestream of type `float64`. The result contains `null`
+              if the value was `null` at that row. Otherwise the row
+              contains the square root the value.
+        """
+        return Timestream._call("sqrt", self)
+
+    def upper(self) -> Timestream:
+        """Return a Timestream with all values converted to upper case."""
+        return Timestream._call("upper", self)
+
+    def zip_max(self, rhs: Union[Timestream, Literal]) -> Timestream:
```

Comment:
SQL calls this `greatest` and `zip_min` is `least`. We probably want to follow suit:

https://www.tutorialsteacher.com/postgresql/greatest-least-function#:~:text=Json%20Type-,PostgreSQL%3A%20GREATEST()%20and%20LEAST(),take%20any%20number%20of%20arguments.